### ISSUE TITLE
HUGE FPS drops with the external storage attached to DrawerController or similar big Storage solutions while requesting items that are not there

### ISSUE THREAD

**Issue by IanDresarie** [refinedmods/refinedstorage#491]
#### Issue description:

When you have a Storage Drawer sytsem connected by an external Storage Bus and try to pull out items that are not there I get huge FPS spikes. Even worse when the Items that I try to pull out are requested by a crafter trying to craft stuff for an exporter with a crafting card.
#### Steps to reproduce:
1. build basic RS network with a crafter and teach it to craft a vanilla furnace
2. connect it to a Storage drawer system
3. set up an export for some crafted Item (in this case furnaces, in my real world it was sugar) using an Exporter with a crafting card
4. provide the storage drawer (input to the controller) with an insufficent supply of items by setting up a cobblegen (I used ExtraUtils stuff without speed upgrades)
5. Watch your FPS spike like crazy when RS tries to pull(/check for) cobble for the requested crafting process...
6. Try it with bigger/smaller Drawer Systems or Chests or other external Storage solutions to see a difference in the impact.

Check out my screenshot for the debug setup I used. In real world the spikes are a lot worse (I got two exporter with crafting cards requesting sugar and a bunch of Ender-IO conduits which are a pain anyway :D) to a point I call unplayable.
![2016-10-21_20 19 57](https://cloud.githubusercontent.com/assets/22986024/19607874/0c4c4808-97cd-11e6-972a-948e41be49ed.png)

Interestingly enough, this also happens with a chest instead of a Drawer system, it's just a lot less intense:
![2016-10-21_20 33 27](https://cloud.githubusercontent.com/assets/22986024/19607969/fa2b6cac-97cd-11e6-9429-8595eac47045.png)

my FPS as a baseline (slightly better FPS than the chest version):
![2016-10-21_20 36 27](https://cloud.githubusercontent.com/assets/22986024/19607997/44419f1e-97ce-11e6-9bc0-1d55abab0fdd.png)

Conclusion: For some reason the way you check external storage for items with you Crafter drops FPS. The bigger the storage it has to check, the bigger the drop.

reals world screenshots:
![2016-10-21_20 43 11](https://cloud.githubusercontent.com/assets/22986024/19608124/4e0dc9f4-97cf-11e6-818f-698246ac0a11.png)
the exporters requesting sugar:
![2016-10-21_20 43 26](https://cloud.githubusercontent.com/assets/22986024/19608126/51d4373a-97cf-11e6-9f3b-ad6765445b94.png)
without the DrawerController (the first 20% of the timeline was before I broke it):
![2016-10-21_20 44 57](https://cloud.githubusercontent.com/assets/22986024/19608143/7afb0472-97cf-11e6-8999-dce9f2192108.png)
(the single big spikes I haven't figures out yet. suspecting something else in Refined STorage that I could not identify yet. They go away when I break the controller, so it has to be something else in RS)
#### Version:
- Minecraft: 1.10.2
- Forge: 12.18.1.2088
- Refined Storage: 1.1.2

Does this issue occur on a server? [not testet]

**Comment by IanDresarie** [comment]
@raoulvdberge I did read the tracker. The only close thing is the Colossal Chest thing which you marked as crossmod. I decided to post my specific results because they indicate it's not a crossmod (which I think means multiple mods interacting) issue and instead one that happens because of RS on it's own.
I am however sorry if I just got confused with the tags and you are already aware of what exactly is happening.
